src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2508,10 +2508,12 @@ internal GlobalExpressionVariable LookupDeclaredField(SyntaxNode node, string id
             else if (hasNames)
             {
                 // We just saw a fixed-position argument after a named argument.
-                if (!hadError)
+                if (!hadError && !Compilation.LanguageVersion.AllowNonTrailingNamedArguments())
                 {
                     // CS1738: Named argument specifications must appear after all fixed arguments have been specified
-                    Error(diagnostics, ErrorCode.ERR_NamedArgumentSpecificationBeforeFixedArgument, argumentSyntax);
+                    Error(diagnostics, ErrorCode.ERR_NamedArgumentSpecificationBeforeFixedArgument, argumentSyntax,
+                        new CSharpRequiredLanguageVersion(MessageID.IDS_FeatureNonTrailingNamedArguments.RequiredVersion()));
+
                     hadError = true;
```

Comment:
>hadError = true; [](start = 20, length = 16)

I think we shouldn't set the has error flag just because there is a mismatch in language version.